### ISSUE TITLE
TODO: Enable toggle grad on/off in torch.compile

### ISSUE THREAD

**Issue by chiahao3** [chiahao3/ptyrad#13]
Currently (0.1.0b10) introduces `torch.compile` that would compile the model and ignores the changes in tensor.requires_grad toggling, so whatever `start_iter` specified in `update_params` would not take any effect (i.e., the reconstruction would follow the initially specified optimizable tensors and we can't add more / delayed onset other optimizable tensors). 

The proper solution would be to recompile the model whenever there's a change in the number of tensors that require grad.

**Comment by chiahao3** [comment]
The latest commit from ptyrad/jit #[feb30db](https://github.com/chiahao3/ptyrad/commit/feb30dbf00bed8dbb19abbea0390777412c1c9a8) added this feature and enable more flexible control with 'end_iter'. I'll merge this into main maybe in a few weeks.

In order to switch on / off grad calculations of optimizable tensors at specific iterations, we'll need to either (1) recompile the `recon_step` at specific iterations or (2) calculate all grads for each params and manually mask them at each iterations. Option 2 is more straightforward but would be spending computation on params that we're not updating. In the commit I went with Option 1.

However, this dynamic control flow (i.e, recompilation at different iterations) also requires some overhead for recompilation (i.e., a 20sec iteration could take 40sec if we're compiling it for the first time), so we should avoid actively recompiling. 

In addution, we also have to set `find_unused_parameters=true` in DDP (multiGPU) otherwise it'll throw errors that some parameters are not participating in the forward/backward. This flag would also add some overhead but is mostly negligible (i.e., roughly 0.1s for a 10sec iteration)

**Comment by chiahao3** [comment]
This has been solved in PtyRAD v0.1.0b11!
